src/services/safe-messages/safeMsgSender.ts

```diff
@@ -94,6 +88,32 @@ export const dispatchSafeMsgConfirmation = async ({
 
   safeMsgDispatch(SafeMsgEvent.CONFIRM_PROPOSE, {
     messageHash,
-    requestId,
   })
 }
+
+const isMessageFullySigned = (message: SafeMessage): message is SafeMessage & { preparedSignature: string } => {
+  return message.confirmationsSubmitted >= message.confirmationsRequired && !!message.preparedSignature
+}
+
+export const dispatchPreparedSignature = async (
+  chainId: string,
+  safeMessageHash: string,
+  onClose: () => void,
+  requestId?: string,
+) => {
+  try {
+    // the response has to be a SafeMessage as it is the only type with safeMessageHash
+    const message = (await getSafeMessage(chainId, safeMessageHash)) as SafeMessage
+
+    if (!isMessageFullySigned(message)) return message
+
+    safeMsgDispatch(SafeMsgEvent.SIGNATURE_PREPARED, {
+      messageHash: safeMessageHash,
+      requestId,
+      signature: message.preparedSignature,
+    })
+    onClose()
```

Comment:
Nit: let's only fetch the message in the `try` and then handle the logic outside of this block, as we do in other dispatchers. Also, why do we need return the `message`?